### PR TITLE
add `cond` builtin

### LINKED ISSUE #77: Native `(cond )` not implemented
## Issue description
The native `(cond ` is not implemented
ie:
```pact
(cond
   (( ...) result-1)
   (( ...) result-2)
  default)

### PR DESCRIPTION
**Pull request by rsoeldner**
closes #77

PR checklist:

* [x] Test coverage for the proposed changes
* [ ] PR description contains example output from repl interaction or a snippet from unit test output
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/kadena-io/pact-core/blob/master/CHANGELOG.md)

Additionally, please justify why you should or should not do the following:

* [ ] Confirm replay/back compat
* [ ] Benchmark regressions
* [ ] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206473309875661

### CODE REVIEW

File: pact/Pact/Core/Serialise/CBOR_V1.hs [hunk 1]
```diff
@@ -820,6 +821,7 @@ instance Serialise CoreBuiltin where
     125 -> pure CoreRoundPrec
     126 -> pure CoreCeilingPrec
     127 -> pure CoreFloorPrec
+    128 -> pure CoreCond
     _ -> fail "unexpeced decoding"
```
Comment (emilypi) [inline pact/Pact/Core/Serialise/CBOR_V1.hs:825]:
```suggestion
    _ -> fail "unexpected decoding"
```

**Review by emilypi — COMMENTED** [review]
LGTM

**Review by 0xd34df00d — APPROVED** [review]

File: pact/Pact/Core/Serialise/CBOR_V1.hs (hunk 1 above)
Comment (jmcardon) [reply pact/Pact/Core/Serialise/CBOR_V1.hs:825]: 👍 agree

**Review by jmcardon — APPROVED** [review]
Approved but please add the test case for https://github.com/kadena-io/pact/issues/1118

**Pull request merged**